src/main/java/com/google/api/codegen/transformer/java/JavaModelTypeNameConverter.java

```diff
@@ -227,6 +231,21 @@ public TypeName getTypeNameForTypedResourceName(
     }
   }
 
+  private static String getResourceNamePackage(ProtoFile file, ResourceNameType resourceNameType) {
```

Comment:
This runs into the problem that packaging is language-dependent. So it will be necessary to have the JavaModelTypeNameConverter, or some other Java-specific class, involved somewhere in the package construction.

Or alternatively, the ResourceNameConfig could provide a getJavaPackage() method, and other languages would add a (for example) getCSharpPackage() method as required. This seems neater that what we have right now, but a bit of a break with the existing patterns.